packages/material-ui/src/Box/styleFunction.js

```diff
@@ -0,0 +1,119 @@
+import {
+  borders,
+  display,
+  flexbox,
+  grid,
+  positions,
+  palette,
+  shadows,
+  sizing,
+  spacing,
+  typography,
+} from '@material-ui/system';
+
+const filterProps = [
+  ...borders.filterProps,
+  ...display.filterProps,
+  ...flexbox.filterProps,
+  ...grid.filterProps,
+  ...positions.filterProps,
+  ...palette.filterProps,
+  ...shadows.filterProps,
+  ...spacing.filterProps,
+  ...typography.filterProps,
+  'sx',
+];
+
+const getThemeValue = (prop, value, theme) => {
```

Comment:
Should we update compose from `material-ui/system` to do something similar? Should we hold off from those changes now?

Reply:
We are only doing this for performance considerations, right?

Reply:
Yes, but to be honest I can't see any perf gains at this moment. Maybe we should just compose the new `sx` `styleFunction` and start from there...